src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -87,6 +88,234 @@ protected override Conversion GetInterpolatedStringConversion(BoundUnconvertedIn
+                analyzedArguments.Clear();
+                overloadResolutionResult.Clear();
+
+                ArrayBuilder<MethodSymbol> candidateMethods;
+                if (part is BoundStringInsert insert)
+                {
+                    if (interpolationHoleCandidates is null && !tryGetCandidateMethods("TryFormatInterpolationHole", ref useSiteInfo, out interpolationHoleCandidates))
+                    {
+                        // PROTOTYPE(interp-string): We'll likely want to continue attempting to bind for errors when we're not directly being converted to a string
+                        free();
+                        builderFormatCalls.Free();
+                        builderArguments = default;
+                        return false;
+                    }
+
+                    candidateMethods = interpolationHoleCandidates;
+                    analyzedArguments.Arguments.Add(insert.Value);
+                    analyzedArguments.Names.Add(null);
+
+                    if (insert.Alignment is not null)
+                    {
+                        analyzedArguments.Arguments.Add(insert.Alignment);
+                        analyzedArguments.Names.Add(SyntaxFactory.IdentifierName("alignment"));
+                    }
+                    if (insert.Format is not null)
+                    {
+                        analyzedArguments.Arguments.Add(insert.Format);
+                        analyzedArguments.Names.Add(SyntaxFactory.IdentifierName("format"));
+                    }
+                }
+                else
+                {
+                    candidateMethods = baseStringCandidates;
+                    analyzedArguments.Arguments.Add(part);
+                }
+
+                _binder.OverloadResolution.MethodInvocationOverloadResolution(
+                    candidateMethods,
+                    typeArguments,
+                    implicitBuilderReceiver,
+                    analyzedArguments,
+                    overloadResolutionResult,
+                    ref useSiteInfo);
+
+                if (!overloadResolutionResult.Succeeded
+                    || overloadResolutionResult.ValidResult.Member.CallsAreOmitted(source.SyntaxTree)
+                    || !returnTypeMatches(overloadResolutionResult.ValidResult.Member, usesBoolReturn))
+                {
+                    // PROTOTYPE(interp-string): We'll likely want to continue attempting to bind for errors when we're not directly being converted to a string
+                    free();
+                    builderFormatCalls.Free();
+                    interpolationHoleCandidates?.Free();
+                    builderArguments = default;
+                    return false;
+                }
+
+                var argsToParam = overloadResolutionResult.ValidResult.Result.ArgsToParamsOpt;
+
+                _binder.CoerceArguments(overloadResolutionResult.ValidResult, analyzedArguments.Arguments, diagnostics);
```

Comment:
>CoerceArguments [](start = 24, length = 15)

It feels like we are doing too much manual work (coercing arguments, binding default arguments, etc.) here and creating a maintenance burden for the future in terms of making sure all the checks that we perform for regular call are also performed here and tested here, etc. I think we should use existing methods in the binder to bind the call into a ```BoundCall``` node based on the set of candidates and given arguments. Then we can deconstruct that node into ```MethodArgumentInfo```, but I am not even sure why would we want to do that instead of simply storing the ```BoudCall``` and using it later in the pipeline.

Reply:
The main issue here is that we don't have a node to act as a receiver for this call. I suppose I could make a placeholder receiver that we then swap out in lowering for the actual builder instance, but we don't do similar tricks for any of the foreach calls.

---
In reply to: [610745021](https://github.com/dotnet/roslyn/pull/52422#discussion_r610745021) [](ancestors = 610745021)

Reply:
> The main issue here is that we don't have a node to act as a receiver for this call. I suppose I could make a placeholder receiver that we then swap out in lowering for the actual builder instance, but we don't do similar tricks for any of the foreach calls.

For each calls are significantly simpler than this.

---
In reply to: [612035214](https://github.com/dotnet/roslyn/pull/52422#discussion_r612035214) [](ancestors = 612035214,610745021)